cmds/dut/doc.go

```diff
@@ -0,0 +1,66 @@
+// dut manages Devices Under Test (a.k.a. DUT) from a host.
+// A primary goal is allowing multiple hosts with any architecture to connect.
+//
+// This program was designed to be used in u-root images, as the uinit,
+// or in other initramfs systems. It can not function as a standalone
+// init: it assumes network is set up, for example.
+//
+// In this document, dut refers to this program, and DUT refers to
+// Devices Under Test. Hopefully this is not too confusing, but it is
+// convenient. Also, please note: DUT is plural (Devices). We don't need
+// to say DUTs -- at least one is assumed.
+//
+// The same dut binary runs on host and DUT, in either device mode (i.e.
+// on the DUT), or in some host-specific mode. The mode is chosen by
+// the first non-flag argument. If there are flags specific to that mode,
+// they follow that argument.
+// E.g., when uinit is run on the host and we want it to enable cpu daemons
+// on the DUT, we run it as follows:
+// dut cpu -key ...
+// the -key switch is only valid following the cpu mode argument.
+//
+// modes
+// dut currently supports 3 modes.
+//
+// The first, default, mode, is "device". In device mode, dut makes an http connection
```

Comment:
is this "In device mode, DUT makes"  vs dut makes?  or dut on DUT makes?

also, should the DUT start the RPC server before calling back to the host?  if you do it as written, there's a window of time where the host dut could call back to the DUT, but the service isn't ready yet.  if you time it poorly, you'll get stuck in tcp limbo or something.